core/templates/components/button-directives/create-activity-button.component.spec.ts

```diff
@@ -0,0 +1,290 @@
+  }));
+
+  it('should initialize and create new exploration if user cannot' +
+    ' create collections', fakeAsync(() => {
+    const userServiceSpy = spyOn(userService, 'getUserInfoAsync')
+      .and.returnValue(Promise.resolve(userInfoCanNotCreateCollection));
+    const urlParamsSpy = spyOn(urlService, 'getUrlParams').and.returnValue({
+      mode: 'create'
+    });
+    const explorationCreationServiceSpy = spyOn(
+      explorationCreationService, 'createNewExploration');
+
+    expect(component.canCreateCollections).toBe(false);
+    expect(component.userIsLoggedIn).toBe(false);
+
+    component.ngOnInit();
+    tick();
+    fixture.detectChanges();
+
+    expect(userServiceSpy).toHaveBeenCalled();
+    expect(urlParamsSpy).toHaveBeenCalled();
+    expect(explorationCreationServiceSpy).toHaveBeenCalled();
+    expect(component.canCreateCollections).toBe(false);
+    expect(component.userIsLoggedIn).toBe(true);
+  }));
+
+  it('should stop the creation process if another' +
+    ' creation is in progress', () => {
+    component.creationInProgress = true;
+    expect(component.initCreationProcess()).toBe();
+  });
+
+  it('should create new exploration if user cannot create collections', () => {
+    component.creationInProgress = false;
+    component.canCreateCollections = false;
+    const explorationCreationServiceSpy = spyOn(
+      explorationCreationService, 'createNewExploration');
+
+    component.initCreationProcess();
+
+    expect(explorationCreationServiceSpy).toHaveBeenCalled();
+  });
+
+  it('should redirect user to create new exploration when user clicks' +
+    ' create button and is not on creator dashboard page', () => {
+    component.creationInProgress = false;
+    component.canCreateCollections = true;
+    const urlServiceSpy = spyOn(urlService, 'getPathname').and.returnValue(
+      'not/creator-dashboard');
+    spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
+      location: {
+        replace: (val: string) => {}
+      }
+    });
+    const replaceSpy = spyOn(windowRef.nativeWindow.location, 'replace');
+
+    component.initCreationProcess();
+
+    expect(urlServiceSpy).toHaveBeenCalled();
+    expect(replaceSpy).toHaveBeenCalledWith('/creator-dashboard?mode=create');
+  });
+
+  it('should open a create activity modal if user' +
+    ' can create collections and is on creator dashboard page', () => {
+    component.creationInProgress = false;
+    component.canCreateCollections = true;
+    const urlServiceSpy = spyOn(urlService, 'getPathname').and.returnValue(
+      '/creator-dashboard');
+    const modalSpy = spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
+      return <NgbModalRef>({
+        result: Promise.resolve('success')
+      });
+    });
+
+    component.initCreationProcess();
+
+    expect(urlServiceSpy).toHaveBeenCalled();
+    expect(modalSpy).toHaveBeenCalled();
+  });
+
+  it('should close modal when user clicks outside of' +
```

Comment:
This is a good test statement but the test doesn't check this. You need to have some kind of check that the modal is actually closed (and a similar check before that verifies that the modal is open before the action is taken).

Reply:
I think, only the success and rejection callback of the modal is needed to be tested (which is happening), but the test description is different. Should I change the test description accordingly or is testing whether a modal is opened or closed the right way to go?

Reply:
As long as the test description and implementation match, and it tests something useful (that could conceivably go wrong and relates to what an external observer sees rather than an implementation detail) then things should be fine.

Reply:
Changed the description, and removed the succesCallback test, as successCallback did nothing.